app/src/main/java/fr/neamar/kiss/UiTweaks.java

```diff
@@ -49,6 +52,40 @@ static void updateThemePrimaryColor(Activity activity) {
         }
     }
 
+    static void applyRoundedCorners(MainActivity mainActivity) {
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(mainActivity);
+
+        if (prefs.getBoolean("pref-rounded-bars", false)) {
+            mainActivity.findViewById(R.id.main_kissbar).setBackgroundResource(R.drawable.rounded_kiss_bar);
+            mainActivity.findViewById(R.id.favoritesBar).setBackgroundResource(R.drawable.rounded_search_bar);
+            mainActivity.findViewById(R.id.searchEditLayout).setBackgroundResource(R.drawable.rounded_search_bar);
+        } else {
+            int kissGreen = mainActivity.getResources().getColor(R.color.kiss_green);
+            mainActivity.findViewById(R.id.main_kissbar).setBackgroundColor(kissGreen);
+
+            // get styled background color
+            int[] attrs = {R.attr.searchBackgroundColor};
```

Comment:
Thinking out loud, but wouldn't it be easier to understand to create another drawable xml and load it here?

Something like

```xml
<?xml version="1.0" encoding="utf-8"?>
<shape xmlns:android="http://schemas.android.com/apk/res/android"
    android:shape="rectangle">
    <solid android:color="?attr/searchBackgroundColor" />
</shape>
```

Reply:
Or even easier, to keep the default background in the XML to what is used to be (`?attr/searchBackgroundColor`) and then load another background resource only if rounded interface is on?

Reply:
I will think about what you said.